datalad/downloaders/base.py

```diff
@@ -104,6 +108,11 @@ def __init__(self, credential=None, authenticator=None):
         self.credential = credential
         self.authenticator = authenticator
         self._cache = None  # for fetches, not downloads
+        self._lock = InterProcessLock(
+            op.join(
+                cfg.obtain('datalad.locations.cache'),
+                'locks',
+                'downloader-auth.lck'))
```

Comment:
Based on the `InterProcessLock` documentation, I didn't see any guarantees that this directory is created if it doesn't exist.  However, looking at the `fasteners` source code (and based on the passing tests), it does seem to be the case.  Okay.

Reply:
yeah -- I figured it "trial&error" way and didn't want to explicitly call `mkdirs` here since lock might not be needed (downloader not actually used) etc.  So decided to stick on fasteners to provide it